### PR TITLE
[WIP] validate-modules: 'choices' disable string conversion

### PR DESCRIPTION
**Pull request by pilou-**
##### SUMMARY
`validate-modules` sanity check: 
* disable string conversion
* allow to explicitly use `None` in `choices`

Fix E328 errors:
* `azure_rm_manageddisk`: `zone` param: values are `str`
* `clc_loadbalancer`: `port` parameter is an `int`
* `yum_repository`: `ip_resolve` parameter: fix doc (values are strings)
* `os_security_group_rule.py`: `protocol` parameter: `None` (Python)/`null` (YAML)

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
validate-modules

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by s-hertel** [comment]
I like this but it seems like it will break any third party modules using null as a choice and relying on it being a string. It looks like there are other cases in the repos as well, like ansible/lib/ansible/modules/network/fortios/fortios_vpn_ipsec_manualkey_interface.py.

**Comment by Fred-sun** [comment]
@pilou- When you are free, please help to solve the conflict in PR and complete the change of PR, I will promote the review! Thank you very much!

**Comment by cloudnull** [comment]
sadly this looks like its in merge conflict and will need a rebase.

**Review by cloudnull — APPROVED** [review]
:+1:

**Comment by Fred-sun** [comment]
kindly ping

**Review by jtyr — APPROVED** [review]
LGTM for `yum_repository`. Btw. you have to resolve the CI/CD issues before this get merged.

**Comment by Fred-sun** [comment]
kindly ping

**Review by sivel — APPROVED** [review]
Conditionally approved pending a rebase

**Comment by Fred-sun** [comment]
@pilou- Are you still here? Could  you help finish PR change and rebase PR to resolve the conflicting files. Once you finish change, I will push for merge

**Comment by pilou-** [comment]
@Fred-sun #64366 was extracted from this PR, can you help merging this one too ?

Yes, I will update this PR.

**Comment by Fred-sun** [comment]
> @Fred-sun #64366 was extracted from this PR, can you help merging this one too ?
> Yes, I will update this PR.


@pilou- I don't have permission to merge， But I will push this! Thank you very much!

**Comment by Fred-sun** [comment]
Hi @haiyuazhang Can you help merge the PR mentioned above?

**Comment by sshnaidm** [comment]
Thanks for submitting patch for Openstack Ansible modules!
We moved Openstack Ansible modules to Openstack repositories.
Next patches should be submitted not with Ansible Github but with
Openstack Gerrit: https://review.opendev.org/#/q/project:openstack/ansible-collections-openstack
Please submit your code there from now.
Thanks for your contribution and sorry for inconvienience.

**Comment by Fred-sun** [comment]
@pilou- Thank you for taking the time to contribute to this PR. We will transfer ansible 's azure module related Issue and PR to azure collection (https://github.com/ansible-collections/azure/pulls), can you transfer the Issue to azure collection repo?

**Pull request closed without merging**
